components/constellation/constellation.rs

```diff
@@ -1368,9 +1368,15 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
     fn handle_subframe_loaded(&mut self, pipeline_id: PipelineId) {
         let (frame_id, parent_id) = match self.pipelines.get(&pipeline_id) {
-            Some(pipeline) => match pipeline.parent_info {
-                Some((parent_id, _)) => (pipeline.frame_id, parent_id),
-                None => return warn!("Pipeline {} has no parent.", pipeline_id),
+            Some(pipeline) => {
+                // If this is the initial about:blank page, no load event should be fired.
+                if pipeline.is_initial_about_blank {
+                    return;
+                }
```

Comment:
What is the behaviour of the following?
* `<iframe>`
* `<iframe src="">`
* `<iframe src="about:blank">`
* `<iframe src="http://42.0x1232131">` (some value that yields a URL parse error, which should use about:blank as a fallback)

Specifically, is exactly one asynchronous load event dispatched in each case?